src/pacmod3_core.cpp

```diff
@@ -460,6 +460,25 @@ void DoorRptMsg::parse(const uint8_t *in)
   fuel_door_open_is_valid = ((in[1] & 0x40) > 0);
 }
 
+void EngineRptMsg::parse(const uint8_t *in)
+{
+  uint16_t temp1, temp2;
+  uint8_t temp3;
+
+  temp1 = ((static_cast<uint16_t>(in[0]) << 8) | in[1])/4;
```

Comment:
I don't think this is going to do what you want. The division by 4 should be where the float conversion comes in and it would be implied. If you leave it as-is, you are dividing an unsigned int by another int (4) and you are going to get an int as a result. What I think you want is to do the bitwise operations in the first step and then divide by 4.0 (a float) which will result in a float.

Reply:
That makes more sense, I'll update those. Thanks!